### PR TITLE
feat: GQL Email (core/email) 

### LINKED ISSUE #5753: [GQL] Email (core/email)
This plugin has one meteor method that retries an email job in the queue.

### Collections

none

### Meteor dependencies

- [ ] meteor/check
- [ ] meteor/meteor (for method definition)

### Methods

- [ ] emails/retryFailed

### File structure
```
.
├── index.js
└── methods
    ├── index.js
    └── retryFailed.js
```

### PR DESCRIPTION
**Pull request by trojanh**
Resolves #5753 
Impact: **major**  
Type: **feature**

## Issue
There is no retryFailed method for Emils

## Solution
Port retryFailed from old codebase

### CODE REVIEW

**Comment by trojanh** [comment]
sure Thanks @mikemurray , I did all those changes, you can test it now, 

I would really appreciate if you could help me with the issue I mentioned [here](https://github.com/reactioncommerce/reaction/pull/5786#issuecomment-552557236), 
I tried restarting the release-3.0 branch a few times from scratch. `develop` branch seems to be working fine but release-3 just isn't. 

Since there's no logs of graphql request on server I am really not sure how to debug this.

**Comment by mikemurray** [comment]
@trojanh Try these env vars. I think you have too many for current 3.0

```
MONGO_URL=mongodb://mongo:27017/reaction
ROOT_URL=http://localhost:3000
```

**Comment by mikemurray** [comment]
@trojanh As for testing, I looked through the plugin and it's not fully set up to expose these mutations to the API. `sendEmail` is an internal mutation so it doesn't have full setup.

If you want this mutation to be exposed in the API, it'll need a mutation resolver and GQL schema.

**Comment by trojanh** [comment]
sure @mikemurray , thanks for the help. 

I was under the impression it would be used internally, so I just created mutation, but now that you've made it clear that it is supposed to be an API, I'll update it.

**Comment by trojanh** [comment]
@mikemurray I have converted the `retryFail` method to Graphql API now, let me know if there's anything else that needs to be done.  Thanks.

File: src/core-services/email/resolvers/Mutation/retryFailedEmail.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * @param {Object} context - an object containing the per-request state
+ * @returns {Promise<Object>} ResendFailedEmailPayload
+ */
+export default async function resendFailedEmail(parentResult, { input }, context) {
+  const {
+    clientMutationId = null,
+    jobId: opaqueJobId,
+    shopId: opaqueShopId
+  } = input;
+  const resp = await context.mutations.resendFailedEmail(context, {
```
Comment (mikemurray) [inline src/core-services/email/resolvers/Mutation/retryFailedEmail.js]: I think you meant `retryFailedEmail` to match your mutation function

**Comment by trojanh** [comment]
@mikemurray I have fixed the docs and method name. Thanks for the feedback.

File: src/core-services/email/mutations/retryFailedEmail.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    await job.ready();
+  } else {
+    // Otherwise rerun the completed job
+    // `rerun` clones the job and returns the id.
+    // We'll set the new one to ready
+    emailJobId = job.rerun();
+  }
+
+  // Set the job status to ready to trigger the Jobs observer to trigger sendEmail
+  backgroundJobs.update({ _id: emailJobId }, {
```
Comment (mikemurray) [inline src/core-services/email/mutations/retryFailedEmail.js]: This should be updating the `Emails` collection. You can get it from context like so.

```
  const { checkPermissions, backgroundJobs, collections } = context;
  const { Emails } = collections;
```

Comment (trojanh) [reply src/core-services/email/mutations/retryFailedEmail.js]: It was earlier used for `backgroundJobs` but thanks for the feedback. I have made the changes now.

**Pull request merged**